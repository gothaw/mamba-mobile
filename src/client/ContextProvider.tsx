import React, { createContext, FunctionComponent } from 'react';

import { ContextInterface } from "../types";
import { theme } from "../theme";

type Props = {
  children: React.ReactNode
};

const Context = createContext<ContextInterface | null>(null);

const ContextProvider: FunctionComponent<Props> = ({children}) => {
  return (
    <Context.Provider value={theme}>
      {children}
    </Context.Provider>
  );
};

export { Context, ContextProvider };
