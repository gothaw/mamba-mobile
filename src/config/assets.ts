export enum Images {
  MenuWeb = require("../assets/img/spider_web.png"),
  MambaLogo = require("../assets/img/mamba_logo_color.png"),
  MenuIcon = require("../assets/img/menu_icon.png"),
  GameBackground = require("../assets/img/game_background.jpg")
}

export enum Fonts {
  LuckiestGuy = require("../assets/fonts/LuckiestGuy-Regular.ttf")
}