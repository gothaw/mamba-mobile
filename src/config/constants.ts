export enum Texts {
  Start = "start",
  Mamba = "mamba"
}

export enum TestIds {
  MainLogo = "main-logo",
  MainMenu = "main-menu",
  Game = "game",
  MenuIcon = "menu-icon"
}