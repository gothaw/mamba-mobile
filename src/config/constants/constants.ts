export enum Text {
  START = "start"
}