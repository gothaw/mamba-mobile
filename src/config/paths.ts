export enum Paths {
  MainMenu = "/",
  Game = "/mamba"
}