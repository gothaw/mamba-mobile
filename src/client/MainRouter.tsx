import React, { FunctionComponent, useContext } from "react";
import { View } from "react-native";
import { NativeRouter, Route, Routes } from "react-router-native";

import Icon from "../components/Icon";
import { Images, Paths } from "../config";
import { TestIds } from "../config/constants";
import Game from "../views/Game";
import MainMenu from "../views/MainMenu";

import { Context } from "./ContextProvider";

interface Props {
}

const MainRouter: FunctionComponent<Props> = () => {

  const context = useContext(Context);

  return (
    <View style={{...context.theme.container}}>
      <NativeRouter>
        <Routes>
          <Route path={Paths.MainMenu} element={<MainMenu/>}/>
          <Route path={Paths.Game} element={<Game/>}/>
        </Routes>
        <Icon
          src={Images.MenuIcon}
          style={context.theme.mainMenuIconWrapper}
          imgStyle={context.theme.mainMenuIconImg}
          testId={TestIds.MenuIcon}
        />
      </NativeRouter>
    </View>
  );
};

export default MainRouter;
