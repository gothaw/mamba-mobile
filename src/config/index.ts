import { Images } from "./assets";
import { Texts } from "./constants";
import { Paths } from "./paths";

export {
  Images,
  Paths,
  Texts
};