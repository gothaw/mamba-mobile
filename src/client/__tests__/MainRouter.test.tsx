import { act, render, waitFor } from "@testing-library/react-native";

import { TestIds } from "../../config/constants";
import { ContextProvider } from "../ContextProvider";
import MainRouter from "../MainRouter";

const setup = () => {
  return render(
    <ContextProvider>
      <MainRouter />
    </ContextProvider>
  );
};

describe("MainRouter", () => {
  let wrapper;
  beforeEach(async () => {
    wrapper = setup();

    await act(async () => {
      await new Promise(resolve => setImmediate(resolve));
    });
  });
  
  it("should render", async () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  describe(("on main menu"), () => {
    it("should display main menu and menu icon", async () => {
      const mainMenu = await wrapper.getByTestId(TestIds.MainMenu);
      const icon = await wrapper.findByTestId(TestIds.MenuIcon);

      await waitFor(() => {
        expect(icon).toBeTruthy();
        expect(mainMenu).toBeTruthy();
      });
    });
  });

  describe(("in game"), () => {
    it("should display game and menu icon", async () => {

      const startBtn = await wrapper.getByTestId("start");

      await act(async () => {
        await new Promise(resolve => setImmediate(resolve));
      });

      // const click = new MouseEvent("press");
      // Object.assign(click, {preventDefault: jest.fn()});

      // fireEvent(startBtn, "press");

      await act(async () => {
        await new Promise(resolve => setImmediate(resolve));
      });

      const game = await wrapper.findByTestId(TestIds.Game);
      const icon = await wrapper.findByTestId(TestIds.MenuIcon);

      await waitFor(() => {
        console.log(startBtn);

        expect(game).toBeTruthy();
        expect(icon).toBeTruthy();
      });
    });

    // it("should go back to", () => {
    //
    // });
  });
});