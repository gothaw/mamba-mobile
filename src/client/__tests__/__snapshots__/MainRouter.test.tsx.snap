// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainRouter should render 1`] = `
<View
  style={
    {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          4290491705,
          4287202352,
        ],
        "endPoint": [
          0.25,
          1,
        ],
        "locations": undefined,
        "startPoint": [
          0,
          0,
        ],
      }
    }
    style={
      {
        "flex": 1,
        "position": "relative",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "position": "relative",
        }
      }
      testID="main-menu"
    >
      <View
        style={
          {
            "left": -300,
            "position": "absolute",
            "top": -400.2,
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 1334,
              "resizeMode": "contain",
              "width": 750,
            }
          }
        />
      </View>
      <View
        style={
          {
            "bottom": -400.2,
            "position": "absolute",
            "right": -300,
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 1334,
              "resizeMode": "contain",
              "width": 750,
            }
          }
        />
      </View>
      <View
        style={{}}
      >
        <Image
          source={1}
          style={
            {
              "height": 667,
              "resizeMode": "contain",
              "width": 375,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#3b8686",
            "borderColor": "#ffe100",
            "borderRadius": 15,
            "borderWidth": 4,
            "elevation": 10,
            "height": "15%",
            "marginTop": 15,
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.34,
            "shadowRadius": 6.27,
            "width": "25%",
          }
        }
      >
        <Text
          style={
            {
              "boxSizing": "border-box",
              "color": "#ffe100",
              "fontFamily": "Luckiest-Guy",
              "fontSize": 63,
              "height": 63,
              "letterSpacing": 1.5,
              "lineHeight": 63,
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
              "textShadowColor": "#000000",
              "textShadowOffset": {
                "height": 1.5,
                "width": -1.5,
              },
              "textShadowRadius": 1,
              "transform": [
                {
                  "translateY": 48.023999999999994,
                },
              ],
            }
          }
        >
          Start
        </Text>
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
  <View
    style={
      {
        "position": "absolute",
        "right": "5%",
        "top": "5%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Image
        source={1}
        style={
          {
            "height": 200.1,
            "resizeMode": "contain",
            "width": 75,
          }
        }
        testID="menu-icon"
      />
    </View>
  </View>
</View>
`;
